### PR TITLE
Test additional K8s versions

### PR DESCRIPTION
**Pull request by dfarrell07**
Test recent patch-releases for the 1.18, 1.19, and 1.20 minor releases.

The recentness of versions available to test are limited by kindest/node
image available.

https://hub.docker.com/r/kindest/node/tags

Only test in the submariner-operator repository for now, to give maximum
coverage for minimal jobs.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

**Comment by dfarrell07** [comment]
Builds on https://github.com/submariner-io/shipyard/pull/487.

File: .github/workflows/consuming.yml [hunk 1]
```diff
@@ -31,6 +32,15 @@ jobs:
           - project: submariner
             cabledriver: wireguard
             deploytool: operator
+            k8s_version: 1.17.17
```
Comment (mkolesnik) [inline .github/workflows/consuming.yml:35]:
```suggestion
```

Seems redundant?

Comment (dfarrell07) [reply .github/workflows/consuming.yml:35]: Here's a PR to demo the failure that results from removing this:

https://github.com/dfarrell07/shipyard/pull/14/checks?check_run_id=2185768163

Comment (skitt) [reply .github/workflows/consuming.yml:35]: Ah, because if it’s *not* specified, we don’t provide a default, so the kind default is used instead (1.20.2).

Comment (dfarrell07) [reply .github/workflows/consuming.yml:35]: Yeah, the strange bit is that we do provide defaults that seem to work in every case except this one. 

If the new E2E GHA feature isn't used at all, the default works:

https://github.com/submariner-io/submariner/runs/2189218309?check_suite_focus=true

That should be caused by the default here:

https://github.com/submariner-io/shipyard/blob/devel/gh-actions/e2e/action.yaml#L7

If no k8s_version flag is passed to clusters.sh (skipping the GHA), it also handles the default correctly:

https://github.com/submariner-io/shipyard/pull/496/checks?check_run_id=2187434544

That should be caused by the default here:

https://github.com/submariner-io/shipyard/blob/devel/scripts/shared/clusters.sh#L6

Comment (dfarrell07) [reply .github/workflows/consuming.yml:35]: I think it's something weird about how GHA matrix includes work in this case. The default in the GHA is not being used which results in [setting `--k8s_version=""`](https://github.com/submariner-io/shipyard/blob/devel/gh-actions/e2e/action.yaml#L45), which I guess overrides the default in clusters.sh with null.

Anyway, can we move ahead? I think if there's a fix to be made, it's pretty low prio and shouldn't be made here.

**Review by mangelajo — APPROVED** [review]

**Review by mkolesnik — APPROVED** [review]

**Pull request merged**